deltachat-ios/Chat/ChatViewController.swift

```diff
@@ -563,20 +563,19 @@ class ChatViewController: UITableViewController, UITableViewDropDelegate {
                 logger.debug(">>> msgChangedObserver: \(String(describing: ui["message_id"]))")
                 if self.dcChat.canSend, let id = ui["message_id"] as? Int, id > 0 {
```

Comment:
we still need an update for `id == 0` which is [emitted for batch changes](https://c.delta.chat/group__DC__EVENT.html#ga0f52cdaad70dd24f7540abda6193cc2d).

also the `canSend` looks wrong to me here, so, in general, it looks to me as if an `else` is missing.

otoh, i would move the `msg.chatId == self.chatId` check up, so that the whole thing is not exectuted for foreign changes - that way, we can also avoid loading the message from the database in most cases.

i can do a commit with these changes, but one question before: why is the DRAFT handling needed here at all? this is a bit unclear to me.

Reply:
IIRC, it's related to webxdc drafts that we initilize with values before sending

Reply:
> we still need an update for `id == 0` which is [emitted for batch changes](https://c.delta.chat/group__DC__EVENT.html#ga0f52cdaad70dd24f7540abda6193cc2d).

that might be the root cause for the bug, we never adapted to that change in core. (I think that behavior didn't exist all the time)

Reply:
the behavior exists forever, but maybe also the bug is there since then :)

for the draft: still a bit unclear to me, the draft is not updated from the core "just so" - otoh, it's fine, we do not have issues with the drafts, so maybe leave that part alone for now :)

Reply:
If you find time for it, please feel free to add any commits here!

Reply:
yip, i'll do it!